src/jobs/jobs.router.js

```diff
@@ -2,18 +2,85 @@ const winston = require('winston');
 const controller = require('./jobs.controller');
 const slackService = require('../slack/slack.service');
 const teamsController = require('../teams/teams.controller');
-const jobService = require('./jobs.service');
+const jobsUtils = require('./jobs.utils');
+const jobsService = require('./jobs.service');
 const Responses = require('../constants/responses');
 
 /**
- * Create a new job job.
- * If the job already exists in the database, just broadcast to the channel
+ * Handle /job post command
  */
 async function post(req, res) {
   winston.info('jobs-router:post', { payload: req.body });
 
+  const slackMessage = req.body.text;
+  const slackCommand = jobsUtils.getCommand(slackMessage);
+
+  return commandFunctions[slackCommand](req, res);
+}
+
+/**
+ * Vote an existing job.
+ */
+async function vote(req, res) {
+  winston.info('jobs-router:vote', { payload: req.body.payload });
+
+  try {
+    const payload = JSON.parse(req.body.payload);
+    const action = payload.actions[0];
+    const type = action.value.toLowerCase();
+    const uid = `${payload.team.id}-${payload.user.id}`;
+    const jobLink = jobsUtils.getLink(payload.original_message.attachments[0].title_link);
+    const jobId = jobsService.hash(jobLink);
+    const job = await controller.vote(jobId, uid, type);
+    const updatedMessage = slackService.serializeJob(job);
+
+    res
+      .status(200)
+      .send(updatedMessage);
+  } catch (error) {
+    winston.error('jobs-router:vote', { payload: req.body.payload, error });
+
+    return res
+      .status(Responses.Jobs.Post.ErrorVote.CODE)
+      .send(Responses.Jobs.Post.ErrorVote.MESSAGE);
+  }
+}
+
+/**
+ * Get all jobs from the databse
+ */
+async function list(req, res) {
+  winston.info('jobs-router:list', { payload: req.body });
+
+  try {
+    const list = await controller.getAll();
+    res.json(list);
+  } catch (error) {
+    winston.error('jobs-router:list', { payload: req.body, error });
+    return res
+      .status(Responses.Jobs.Post.ErrorList.CODE)
+      .send(Responses.Jobs.Post.ErrorList.MESSAGE);
+  }
+}
+
+const commandFunctions = {
+  help:    _sendHelp,
+  joblist: _sendJobList,
+  empty:   _sendEmpty,
+  createjob:  _sendCreateJob,
+  wrongcommand: _sendHelp
+};
+
+
+/**
+ * Create a new job job.
+ * If the job already exists in the database, just broadcast to the channel
+ */
+async function _sendCreateJob(req, res) {
```

Comment:
Im not sure about the `send` prefix, could just name them: createJob, help, list and error?